Realm/RLMRealmUtil.mm

```diff
@@ -18,29 +18,51 @@
 
 #import "RLMRealmUtil.hpp"
 
-#import "RLMObjectSchema_Private.hpp"
+#import "RLMAsyncTask_Private.h"
 #import "RLMObservation.hpp"
+#import "RLMRealmConfiguration_Private.hpp"
 #import "RLMRealm_Private.hpp"
 #import "RLMUtil.hpp"
 
-#import <Realm/RLMConstants.h>
-#import <Realm/RLMSchema.h>
-
-#import <realm/obj.hpp>
 #import <realm/object-store/binding_context.hpp>
+#import <realm/object-store/impl/realm_coordinator.hpp>
 #import <realm/object-store/shared_realm.hpp>
 #import <realm/object-store/util/scheduler.hpp>
 
 #import <map>
-#import <mutex>
 
 // Global realm state
-static auto& s_realmCacheMutex = *new std::mutex();
+static auto& s_realmCacheMutex = *new RLMUnfairMutex;
 static auto& s_realmsPerPath = *new std::map<std::string, NSMapTable *>();
 static auto& s_frozenRealms = *new std::map<std::string, NSMapTable *>();
 
-void RLMCacheRealm(std::string const& path, void *key, __unsafe_unretained RLMRealm *const realm) {
-    std::lock_guard<std::mutex> lock(s_realmCacheMutex);
+static void *cacheKeyForOptions(RLMConfinement const& options) {
+    if (auto queue = options.queue) {
+        if (queue != dispatch_get_main_queue()) {
+            return (__bridge void *)queue;
+        }
+    }
+    else if (id actor = options.actor) {
+        // check for main actor?
+        return (__bridge void *)actor;
```

Comment:
How are we checking actor equality?

Reply:
Actors are reference types, so pointer equality works fine.